### PR TITLE
Add documentation to external API

### PR DESCRIPTION
**Pull request by malliaridis**
Adds documentation to external API (untested).

I couldn't find a way to test the documentation web rendering like in https://stream-labs.github.io/streamlabs-obs-api-docs/docs/index.html, so the changes couldn't be tested yet properly.

I also found various bugs and issues while creating this documentation and I will contribute some bug fixes in different PRs, so stay tuned. :)

I also found out that the page https://stream-labs.github.io/streamlabs-obs-api-docs/docs/index.html is different from https://stream-labs.github.io/streamlabs-obs-api-docs/docs/. What exactly happens there? Are both related to this project or to https://github.com/stream-labs/streamlabs-obs-api-docs?

I have some wishes for the future too:
- Please provide information on how to contribute to this project properly, e.g. where to do the merge requests, what information you need in PRs, what to avoid
- Please provide information on how the documentation is rendered in https://stream-labs.github.io/streamlabs-obs-api-docs/docs/index.html
- Fix the https://tracker.streamlabs.com link or even better
- Enable GitHub issues
- Provide contact options for questions, discussions and issues related to the development

### CODE REVIEW

File: app/services/api/external-api/scenes/scene.ts [hunk 1]
```diff
@@ -45,56 +57,95 @@ export class Scene implements ISceneModel {
...
-  getNode(sceneNodeId: string): SceneNode {
+  /**
+   * Returns a specific scene node with id {@param sceneNodeId}.
+   *
+   * @param sceneNodeId The id of the scene node
+   * @returns SceneNode with the provided id or `null` if no scene node found
+   */
+  getNode(sceneNodeId: string): SceneNode | null {
     const node = this.scene.getNode(sceneNodeId);
+    if (!node) return null;
```
Comment (malliaridis) [inline app/services/api/external-api/scenes/scene.ts:68]: A small "bugfix" slipped into this PR too. ^_^'

**Comment by andycreeth** [comment]
Hey, thanks for your submission!  I'm going to have our resident external-api expert look this over, but it looks extremely extensive and high quality.  He can answer your questions about how to compile the documentation.  Thank you!  This submission is above and beyond and we really appreciate your hard work.

Also thank you for for your suggestions about the future.  One of of our long term goals is to make it easier to accept outside contributions from developers such as yourself.  I know we're behind on having good documentation and guidelines for contributors, and we will definitely address that in the future.  We are a very small team maintaining a very big app so we have to prioritize things carefully.

For issue tracking, we use an internal system that is not publicly available.  This is to protect the privacy of our users, as our bug reports can contain extensive information.  At some point it would be great to have a separate, publicly accessible tracker, but we're not in a position to that currently.

**Review by holiber — APPROVED** [review]

**Comment by malliaridis** [comment]
@holiber is there a way to view the generated output of the documentation like in https://stream-labs.github.io/streamlabs-obs-api-docs/docs/index.html? Because I added doc comments, but I couldn't validate the rendering of special notation.

**Pull request merged**